### DISCUSSION TITLE
Fixing failure to create a remote repository on remote macOS computer (having Borg via Homebrew) from Vorta

### DISCUSSION THREAD

**Discussion by C-Duv** [borgbase/vorta#2162] [General]
_TL;DR: When using Homebrew's Borg on macOS as a remote server (to host the Borg repository), add `--remote-path /opt/homebrew/bin//borg` as _Extra Borg Arguments_._

I wanted to backup my computer (running Linux: let's call it the client) to another computer (running macOS: let's call it the server) using Vorta.

I've installed Borg on the macOS/server using [this tap](https://github.com/borgbackup/homebrew-tap) (listed on [Borg's documentation](https://borgbackup.readthedocs.io/en/stable/installation.html#distribution-package)) and checked `borg --version` was working.

But, on the Linux/client when trying to create a repository on the macOS/server via Vorta the add failed without any clear message.

Running Vorta form CLI gave more verbose output highlighting some issue to run Borg via SSH:

> Remote: zsh:1: command not found: borg
> Connection closed by remote host. Is borg working on the server

Full log:

```
2024-12-04 18:14:11,026 - vorta.keyring.abc - DEBUG - Only available on macOS
2024-12-04 18:14:11,027 - asyncio - DEBUG - Using selector: EpollSelector
2024-12-04 18:14:11,045 - vorta.keyring.abc - DEBUG - Using VortaSecretStorageKeyring
2024-12-04 18:14:11,045 - vorta.borg.borg_job - DEBUG - Using VortaSecretStorageKeyring keyring to store passwords.
2024-12-04 18:14:11,046 - asyncio - DEBUG - Using selector: EpollSelector
2024-12-04 18:14:11,046 - vorta.keyring.secretstorage - DEBUG - Found 0 passwords matching repo URL.
2024-12-04 18:14:11,047 - vorta.borg.borg_job - DEBUG - Password not found in primary keyring. Falling back to VortaDBKeyring.
2024-12-04 18:14:11,048 - vorta.borg.jobs_manager - DEBUG - Add job for site default
2024-12-04 18:14:11,048 - vorta.borg.jobs_manager - DEBUG - Start job on site: default
2024-12-04 18:14:11,066 - vorta.borg.borg_job - INFO - Running command /usr/bin/borg init --info --log-json --encryption=repokey-blake2 ssh://user_name@remote_server/~/borg_backup
2024-12-04 18:14:11,488 - vorta.borg.borg_job - WARNING - Remote: zsh:1: command not found: borg
2024-12-04 18:14:11,490 - vorta.borg.borg_job - ERROR - Connection closed by remote host. Is borg working on the server?
2024-12-04 18:14:11,631 - vorta.borg.jobs_manager - DEBUG - Finish job for site: default
2024-12-04 18:14:11,631 - vorta.borg.jobs_manager - DEBUG - No more jobs for site: default
```

As I know Borg is installed and working on macOS/server I tried various commands:

* `ssh user_name@remote_server borg --version` ⤇ "zsh:1: command not found: borg"
* `ssh user_name@remote_server which borg` ⤇ "borg not found"
* `ssh user_name@remote_server type borg` ⤇ "borg not found"
* `ssh user_name@remote_server -t 'source /etc/zshrc && borg --version' ⤇ No output at all

Because I know where Homebrew installed borg (`/opt/homebrew/bin//borg`) I can give it and this time it works:

`ssh user_name@remote_server /opt/homebrew/bin//borg --version` ⤇ "borg 1.4.0"

Vorta not able to SSH-execute `borg` because it relies on $PATH working fine is not a Vorta bug, maybe something that could be improved (don't know how as I failed to find a simple command to detect the borg location on the macOS/remote).

So I'm leaving this here:

Adding `--remote-path /opt/homebrew/bin//borg` in the new repository window's _Extra Borg Arguments_field fixed the issue for me.

```
2024-12-04 18:26:06,160 - vorta.borg.borg_job - INFO - Running command /usr/bin/borg init --remote-path /opt/homebrew/bin//borg --info --log-json --encryption=repokey-blake2 ssh://user_name@remote_server/~/borg_backup
2024-12-04 18:26:06,731 - vorta.borg.borg_job - INFO - Initializing repository at "ssh://user_name@remote_server/~/borg_backup"
2024-12-04 18:26:06,753 - vorta.borg.borg_job - INFO - Key in "<RemoteRepository ssh://user_name@remote_server/~/borg_backup>" created.
2024-12-04 18:26:06,754 - vorta.borg.borg_job - INFO - Keep this key safe. Your data will be inaccessible without it.
2024-12-04 18:26:06,878 - vorta.borg.borg_job - INFO - Synchronizing chunks cache...
2024-12-04 18:26:06,881 - vorta.borg.borg_job - INFO - Archives: 0, w/ cached Idx: 0, w/ outdated Idx: 0, w/o cached Idx: 0.
2024-12-04 18:26:06,881 - vorta.borg.borg_job - INFO - Done.
```

Edit: Using Vorta v0.8.12 on Linux Mint 22 ("Wilma")

**Comment by m3nu** [comment]
This option isn't properly saved right now. You see it's not in the command.

This is already fixed in the master branch and will be released soon.

**Reply by C-Duv** [reply]
I don't understand what you mean by "properly saved".

As I do see it in the log I posted I can only assume you are talking about "command" of another place.

Forgot to tell I am using v0.8.12 which is available in the base repositories for my Linux Mint (I updated my first post to add these informations), I'll check for the newer version in backports or similar.

**Reply by m3nu** [reply]
It means there is a bug in the code and the input there isn't saved. Already fixed in `master` branch. I apologize for the confusion this bug has caused you.
